src/loader.js

```diff
@@ -180,23 +181,30 @@ export async function ${dataFetchingFunction}(context) {
    */
   const content = await Markdoc.transform(ast, cfg);
 
+  // Removes undefined
+  return JSON.parse(
+    JSON.stringify({
+      markdoc: {
+        content,
+        frontmatter,
+        file: {
+          path: filepath
+        }
+      },
+    })
+  );
+}
+
+${appDir ? '' : `export async function ${dataFetchingFunction}(context) {
   return {
-    // Removes undefined
-    props: JSON.parse(
-      JSON.stringify({
-        markdoc: {
-          content,
-          frontmatter,
-          file: {
-            path: filepath
-          }
-        },
-      })
-    ),
+    props: await getProps(context),
   };
-}
+}`}
+
+${appDir ? `export const metadata = frontmatter.metadata;` : ''}
```

Comment:
I like this. What if we just _always_ export the full frontmatter?
```suggestion
export {frontmatter}
```

Reply:
The `metadata` named export is a built-in Next.js feature for defining page metadata, so it needs to be exported as `metadata`. Did you mean this maybe?

```js
export const metadata = frontmatter
```

Reply:
Oh I see why you did it this way now. I like this idea, but I want to keep this change isolated from the rest of the work, since introducing the `metadata` field in `frontmatter` with special behavior needs to be documented, and I want to consider it separately. 

Can you extract that change into a separate PR? Thanks so much!